test/app/components/custom_test_controller_component.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class CustomTestControllerComponent < ViewComponent::Base
+  self.test_controller = "CustomTestControllerController"
+
+  def call
+    helpers.foo
+  end
+end
```

Comment:
I think we need some more test cases that make sure the behavior of setting `test_controller` is as expected. I'm pretty sure this test could/would still pass if test_controller was a global parameter vs scoped to the component class. I think a few other things are needed like.

- setting the test_controller doesn't impact other components
- Subclassing a component with test_controller set inherits the test_controller
- Subclassing a component and overriding the test_controller of the super class that set test_controller

Reply:
Done, done, and done @jonspalmer! I've added a new `test_test_controller_as_class_attribute` case to `view_component_test.rb` that should cover these scenarios.